src/metaswitch/clearwater/plugin_tests/test_cassandra_failed_plugin.py

```diff
@@ -0,0 +1,86 @@
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+import unittest
+import mock
+import logging
+
+_log = logging.getLogger()
+
+from clearwater_etcd_plugins.clearwater_cassandra.cassandra_failed_plugin import CassandraFailedPlugin
+from metaswitch.clearwater.cluster_manager.plugin_base import PluginParams
+from metaswitch.clearwater.cluster_manager import alarm_constants
+from metaswitch.clearwater.cluster_manager.plugin_utils import WARNING_HEADER
+
+class TestCassandraFailedPlugin(unittest.TestCase):
+    # run_command returns 0 if a command completes successfully, but python mocks
+    # return 'True', i.e. 1. Force return value of 0 to simulate successes.
+    @mock.patch('clearwater_etcd_plugins.clearwater_cassandra.cassandra_failed_plugin.run_command',\
+                return_value=0)
+    @mock.patch('metaswitch.common.alarms.alarm_manager.get_alarm')
+    # The plugin uses check_output to get the latency value, so we return 100000.
+    @mock.patch('clearwater_etcd_plugins.clearwater_cassandra.cassandra_failed_plugin.subprocess.check_output',\
+                return_value="UN  10.0.0.1   177.36 KB  256     100.0%            92a674aa-a64b-4549-b150-596fd466923f  RAC1")
+    def test_cassandra_failed_leaving_cluster(self,\
+                                              mock_check_output,\
+                                              mock_get_alarm,\
+                                              mock_run_command):
+        """Test the cassandra_failed_plugin leaving cluster process"""
+
+        # Create a plugin with dummy parameters
+        plugin = CassandraFailedPlugin(key="etcd_key/etcd_cluster_key/clustering/cassandra",
+                                       ip='10.0.0.1')
+
+        # Build a cluster_view that includes all possible node states
```

Comment:
I agree, if we're not doing anything with it, there's no point passing in a full cluster view.